transport/transport.go

```diff
@@ -105,18 +102,19 @@ func (b *recvBuffer) load() {
 	if len(b.backlog) > 0 {
 		select {
 		case b.c <- b.backlog[0]:
-			b.backlog[0] = nil
+			b.backlog[0].data = nil
```

Comment:
@mehrdada done for first comment, sorry for delay.

> Also, do we ever close the channel in recvMsg?

I don't think we do explicitly close this channel, but I think it's ok for this channel, since the blocking read can unblock when the context is `Done`